### PR TITLE
Simplify shared dataloader based on recent update on mxnet dataloader

### PR DESCRIPTION
**Pull request by szhengac**
## Description ##
Update SharedDataLoader based on https://github.com/apache/incubator-mxnet/pull/12177

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [ ] https://github.com/apache/incubator-mxnet/issues/12197 is fixed

### Changes ###
- Simplify _ShardedMultiWorkerIter

## Comments ##
This must not be merged before apache/incubator-mxnet#12197 is fixed as we otherwise have code that relies on mxnet<=1.3.0b20180810 and code that relies on mxnet>=1.3.0b20180815

### CODE REVIEW

File: gluonnlp/data/dataloader.py [hunk 1]
```diff
@@ -197,4 +126,5 @@ def _same_process_iter():
 
         # multi-worker
         return _ShardedMultiWorkerIter(self._num_workers, self._dataset,
```
Comment (leezu) [inline gluonnlp/data/dataloader.py]: Is it possible to remove `_ShardedMultiWorkerIter` and just call `_MultiWorkerIter` with the appropriate worker loop?

Comment (szhengac) [reply gluonnlp/data/dataloader.py]: Oh, you are right. We can do it.

**Comment by szha** [comment]
@szhengac do the tests need update?

**Comment by szhengac** [comment]
@szha No, tests do not need update. It seems that the mxnet upstream update is not included in mxnet==1.3.0b20180814

**Comment by leezu** [comment]
This must not be merged before https://github.com/apache/incubator-mxnet/issues/12197 is fixed as we otherwise have code that relies on `mxnet<=1.3.0b20180810` and code that relies on `mxnet>=1.3.0b20180815`

**Comment by mli** [comment]
Job PR-285/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-285/5/index.html

**Pull request merged**